cmd/prysmctl/withdrawal/withdrawal.go

```diff
@@ -0,0 +1,166 @@
+	}
+	u, err := url.ParseRequestURI(beaconNodeHost)
+	if err != nil {
+		return errors.Wrap(err, "invalid format, unable to parse url")
+	}
+	if u.Scheme == "" || u.Host == "" {
+		return fmt.Errorf("provided url %s is not in the format of http(s)://host:port", beaconNodeHost)
+	}
+	foundFilePaths, err := findWithdrawalFiles(c.String(PathFlag.Name))
+	if err != nil {
+		return errors.Wrap(err, "failed to find withdrawal files")
+	}
+	au := aurora.NewAurora(true)
+	if c.Bool(SkipPromptsFlag.Name) {
+		fmt.Println(au.Red("===============IMPORTANT==============="))
+		fmt.Println(au.Red("All prompts have been skipped by providing the skip-prompt flag."))
+		fmt.Println(au.Red("User has agreed to all terms of service and will accept data as is without prompt verification."))
+	} else {
+		fmt.Println(au.Red("===============IMPORTANT==============="))
+		fmt.Println(au.Red("Please read the following carefully"))
+		fmt.Print("This action will allow the partial withdraw of amounts over the 32 staked eth in your active validator balance. \n" +
+			"You will also be entitled to the full withdrawal of the entire validator balance if your validator has exited. \n" +
+			"The partial and full withdrawal . \n" +
+			"Please navigate to our website and make sure you understand the full implications of setting your withdrawal address. \n")
+		fmt.Println(au.Red("THIS ACTION WILL NOT BE REVERSIBLE ONCE INCLUDED. "))
+		fmt.Println(au.Red("You will NOT be able to change the address again once changed. "))
+	}
+
+	setWithdrawalAddressJsons := make([]*apimiddleware.SignedBLSToExecutionChangeJson, 0)
+	for _, foundFilePath := range foundFilePaths {
+		b, err := os.ReadFile(filepath.Clean(foundFilePath))
+		if err != nil {
+			return errors.Wrap(err, "failed to open file")
+		}
+		var to []*apimiddleware.SignedBLSToExecutionChangeJson
+		if err := json.Unmarshal(b, &to); err != nil {
+			log.Warnf("provided file: %s, is not a list of signed withdrawal messages", foundFilePath)
+			continue
+		}
+		setWithdrawalAddressJsons = append(setWithdrawalAddressJsons, to...)
+	}
+	if len(setWithdrawalAddressJsons) == 0 {
+		return errors.New("the list of signed requests is empty")
+	}
+	if !c.Bool(SkipPromptsFlag.Name) {
+		for _, jsonOb := range setWithdrawalAddressJsons {
+			if err := verifyWithdrawalCertainty(r, jsonOb); err != nil {
+				return err
+			}
+		}
+	}
+	return callWithdrawalEndpoints(ctx, beaconNodeHost, setWithdrawalAddressJsons)
+}
+
+func verifyWithdrawalCertainty(r io.Reader, request *apimiddleware.SignedBLSToExecutionChangeJson) error {
+	au := aurora.NewAurora(true)
+	withdrawalConfirmation := request.Message.ToExecutionAddress
+	fmt.Println(au.Red("===================================="))
+	fmt.Println("YOU ARE ATTEMPTING TO SET A WITHDRAWAL ADDRESS TO  " + au.Red(request.Message.ToExecutionAddress).String() + " FOR VALIDATOR INDEX" + au.Red(request.Message.ValidatorIndex).String() + ". ")
+	_, err := withdrawalPrompt(withdrawalConfirmation, r)
```

Comment:
Can we remove the prompt, and just have validation that the flag(s) meaning "I really want to do this and I accept TOS" has been set? I would like to avoid interactive prompts in our code if at all possible.

Reply:
haha 3 people have commented on this I'll make the change then.